pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1276,11 +1281,11 @@ func (qjm *XController) ScheduleNext() {
 
 // Update AppWrappers in etcd
 // todo: This is a current workaround for duplicate message bug.
-func (cc *XController) updateEtcd(currentAppwrapper *arbv1.AppWrapper, caller string) (*arbv1.AppWrapper, error) {
+func (cc *XController) updateEtcd(ctx context.Context, currentAppwrapper *arbv1.AppWrapper, caller string) (*arbv1.AppWrapper, error) {
 	klog.V(4).Infof("[updateEtcd] trying to update '%s/%s' called by '%s'", currentAppwrapper.Namespace, currentAppwrapper.Name, caller)
 	currentAppwrapper.Status.Sender = "before " + caller // set Sender string to indicate code location
 	currentAppwrapper.Status.Local = false               // for Informer FilterFunc to pickup
-	updatedAppwrapper, err := cc.arbclients.ArbV1().AppWrappers(currentAppwrapper.Namespace).Update(currentAppwrapper)
+	updatedAppwrapper, err := cc.arbclients.McadV1beta1().AppWrappers(currentAppwrapper.Namespace).Update(ctx, currentAppwrapper, metav1.UpdateOptions{})
```

Comment:
I am curious to know what would it take to graduate `McadV1beta1()` to `McadV1`

Reply:
@astefanutti keep me honest, but I think that the process would be to change the version of the API in the `api` package, regenerated the code and manually update everything to the new "name". It seems that is the process for using the generated clients. @asm582 did it answer your question ?